### PR TITLE
Support passing in dependencies directly via -f json -d file.json

### PR DESCRIPTION
**Pull request by chicagogrooves**
With example.json containing:
`{"a":["sub/b"],"d":[],"fancy-main/not-index":[],"sub/b":["sub/c"],"sub/c":["d"]}`

New feature:
`./bin/dependo -f json -d example.json > example/index.html ; open example/index.html`
Regression tested with:
`./bin/dependo -f cjs lib/ > example/index.html ; open example/index.html`

### CODE REVIEW

File: bin/dependo [hunk 1]
```diff
@@ -5,24 +5,32 @@
...
+    .option('-f, --format <name>', 'format to parse (amd/cjs/es6/json)', 'amd')
     .option('-x, --exclude <regex>', 'a regular expression for excluding modules')
+    .option('-d, --deps <filename>', 'when used with \'-f json\', a file to load deps from in {p1:[\'c1\',\'c2\']} format')
     .parse(process.argv);
 
-if (!Commander.args.length) {
-    console.log(Commander.helpInformation());
-    process.exit(1);
+
+if (Commander.format !== "json" && !Commander.args.length) {
```
Comment (auchenberg) [inline bin/dependo]: Why is there this special check for ``Commander.format !== "json"``?

**Comment by auchenberg** [comment]
Looks good. I'm a but unsure about the above if-condition, once resolved. I'll merge, and make a new release.

**Comment by chicagogrooves** [comment]
Because in json format you pass the file containing the deps data, you don't need spider files, making the CLI arg required only for the previous use cases.

**Comment by auchenberg** [comment]
We agree that to pass the file you need to pass the file via ``--deps`` as ``dependo -f json --deeps <file>``, right?

The above condition is to display a nice help message if no arguments are passed like ``dependo``. What case does the current condition cover? If I pass ```-f json```, ``Commander.args`` anyway has a length?

**Comment by chicagogrooves** [comment]
We agree on that. Would you rather omit --deps and read the deps data from the first CLI arg ? I don't think you're using the term arguments the same way Commander does. In Commander, -f json is not an arg, AFAIK

**Comment by auchenberg** [comment]
Totally makes sense to use pass the file as the "input" in first argument. :+1:

**Comment by chicagogrooves** [comment]
Cool- were on a roll !

**Pull request merged**